File: Assignment2/StudentRecord.cpp
#include<iostream>
#include <string>
#include <fstream>
using namespace std;

const int MAX_STUDENTS = 100;
struct StudentInfo {
    string name;
    int age;
    string department;
    string year;

};

void ReadStudentData(StudentInfo& student) {
    cout << "Enter student name: ";
    getline(cin, student.name);
    cout << "Enter student age: ";
    cin >> student.age;
    cin.ignore();
    cout << "Enter student department: ";
    getline(cin, student.department);
    cout << "Enter student year: ";
    getline(cin, student.year);
}

void PrintStudentData(const StudentInfo& student) {
    cout << "Student Name: " << student.name << endl;
    cout << "Student Age: " << student.age << endl;
    cout << "Student Department: " << student.department << endl;
    cout << "Student Year: " << student.year << endl;
}

void saveStudentData(const StudentInfo student[], int numStudents) {
    ofstream outfile("student_data.csv");

    if (outfile.is_open()) {
        outfile << "Name,\tAge,\tDepartment,\tYear\n";
        for (int i = 0; i < numStudents; ++i) {
            outfile << student[i].name << ",\t" << student[i].age << ",\t" << student[i].department << ",\t"
            << student[i].year<< endl;
        }
        outfile.close();
        cout << "Student data saved to file." << endl;
    } else {
        cout << "Unable to open file." << endl;
    }
}



int main() {
    StudentInfo students[MAX_STUDENTS];
    int numStudents;
    cout << "Enter the number of students: ";
    cin >> numStudents;
    cin.ignore();
    for (int i = 0; i < numStudents; i++) {
        cout << "Enter information for student " << i + 1 << ":" << endl;
        ReadStudentData(students[i]);

    }
    cout << endl << "Student Information:" << endl;
    for (int i = 0; i < numStudents; i++) {
        cout << "Student " << i + 1 << ":" << endl;
        PrintStudentData(students[i]);
        cout << endl;
    }
    saveStudentData(students, numStudents);



    return 0;
}

File: Assignment6/StudentRecord.java
import java.util.Scanner;

public class StudentRecord {
    String name;
    int age;
    String department;
    int year;

    public void readData(String name, int age, String department, int year) {
        this.name = name;
        this.age = age;
        this.department = department;
        this.year = year;
    }

    public void printData() {
        System.out.println("Name: " + name);
        System.out.println("Age: " + age);
        System.out.println("Department: " + department);
        System.out.println("Year: " + year);
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter the number of students: ");
        int numberOfStudents = scanner.nextInt();
        scanner.nextLine();
        StudentRecord[] students = new StudentRecord[numberOfStudents];

        for (int i = 0; i < numberOfStudents; i++) {
            StudentRecord student = new StudentRecord();
            System.out.println("\nEnter details for student " + (i + 1) + ":");
            System.out.print("Name: ");
            String name = scanner.nextLine();
            System.out.print("Age: ");
            int age = scanner.nextInt();
            scanner.nextLine();
            System.out.print("Department: ");
            String department = scanner.nextLine();
            System.out.print("Year: ");
            int year = scanner.nextInt();
            scanner.nextLine();
            student.readData(name, age, department, year);
            students[i] = student;
        }

        System.out.println("\nStudent Details:");
        for (int i = 0; i < numberOfStudents; i++) {
            System.out.println("\nDetails of student " + (i + 1) + ":");
            students[i].printData();
        }

        scanner.close();
    }
}
